extensions/identity-hub/src/main/java/org/eclipse/dataspaceconnector/identityhub/IdentityHubExtension.java

```diff
@@ -49,12 +61,28 @@ public void initialize(ServiceExtensionContext context) {
         methodProcessorFactory.register(COLLECTIONS_WRITE, new CollectionsWriteProcessor(identityHubStore));
         methodProcessorFactory.register(FEATURE_DETECTION_READ, new FeatureDetectionReadProcessor());
 
-        var identityHubController = new IdentityHubController(methodProcessorFactory);
+        var selfDescription = loadJsonFile(context.getTypeManager().getMapper(), context.getConfig().getString(SELF_DESCRIPTION_DOCUMENT_PATH_SETTING));
+        var identityHubController = new IdentityHubController(methodProcessorFactory, selfDescription);
         webService.registerResource(identityHubController);
     }
 
     @Provider(isDefault = true)
     public IdentityHubStore identityHubStore() {
         return new IdentityHubInMemoryStore();
     }
+
+    /**
+     * Load content from a static JSON file.
+     *
+     * @param path Path to the static file.
+     * @return JSON file content.
+     */
+    private static JsonNode loadJsonFile(ObjectMapper mapper, String path) {
+        try {
+            var content = Files.readString(Path.of(path));
```

Comment:
As the self description is just a static file at this stage, why not just adding the file to the extension JAR and load it here from the classpath?

Reply:
As discussed, we keep the EDC setting enabling to provide a custom self-description. 
If this setting is not specified, then a static self-description from the classpath is used.